src/services/CertificateGenerationService.ts

```diff
@@ -49,42 +49,58 @@ class CertificateGenerationService {
      */
     public async generateCertificate(testResult: any): Promise<IGeneratedCertificateResponse> {
         const config: IMOTConfig = this.config.getMOTConfig();
+        const iConfig: IInvokeConfig = this.config.getInvokeConfig();
         const testType: any = testResult.testTypes;
-        const payload: any = await this.generatePayload(testResult);
+        const payload: any = JSON.stringify(await this.generatePayload(testResult));
         const certificateTypes: any = {
             pass: config.documentNames.vtp20,
             fail: config.documentNames.vtp30,
             prs: config.documentNames.psv_prs
         };
 
-        const reqParams: OptionsWithUri = {
-            method: "POST",
-            uri: `${config.endpoint}/${certificateTypes[testType.testResult]}`,
-            body: payload,
-            headers: {
-                "x-api-key": config.api_key
-            },
-            json: true
+        const invokeParams: any = {
+            FunctionName: iConfig.functions.certGen.name,
+            InvocationType: "RequestResponse",
+            LogType: "Tail",
+            Payload: JSON.stringify({
+                httpMethod: "POST",
+                pathParameters: {
+                    documentName: certificateTypes[testType.testResult],
+                    documentDirectory: config.documentDir
+                },
+                json: true,
+                body: payload
+            }),
         };
 
-        return rp(reqParams)
-        .then((response: string) => {
-            const responseBuffer: Buffer = Buffer.from(response, "base64");
+        return this.lambdaClient.invoke(invokeParams)
+            .then((response: PromiseResult<Lambda.Types.InvocationResponse, AWSError>) => {
+                const res: string = JSON.stringify(response);
+                console.log(`Output Response: ${res}`);
```

Comment:
These console logs are temporary as i am trying to retrieve the output for mocks. Will revert these later.